### ISSUE TITLE
FAIL - Test Failed

### ISSUE THREAD

**Issue by Caztiel** [pkt-cash/pktd#227]
When executed ./do

![image](https://user-images.githubusercontent.com/35174212/99876825-7e546300-2c1f-11eb-826f-814c2a3ec254.png)

**Comment by johnsonjh** [comment]
Do you have additional scroll back? Unfortunately that screen-shot shows only passing tests.

**Comment by Caztiel** [comment]
@johnsonjh Edited to post screenshots showing exact issue.

**Comment by johnsonjh** [comment]
Is this from the `master` or the `pktd-v1.2.x` branch (or release)? 

Could you provide the output of `go version` and `go env`? (You should be using Go 1.14.1 at a minimum, but I'd recommend using 1.15.5 currently.)

Also, what architecture and OS? A 64-bit architect? If it's Linux, can you also give the output of `cat /etc/*elease` and `lsb_release -a`? 

Also, if a 64-bit Linux, is this a distribution provided toolchain? If this **is** a distribution provided toolchain, can you reproduce the test failure with the [official release](https://storage.googleapis.com/golang/getgo/installer_linux)?

Sorry to ask for so much detail, but it's a test failure I've not encountered and one strange enough that I want to fully understand your build environment.

**Comment by johnsonjh** [comment]
@cjdelisle Verified with @Caztiel that whatever the root cause was (bad test or actual deadlock), it's already fixed in `develop`.

**Comment by johnsonjh** [comment]
@cjdelisle safe to close

**Comment by Frank-Buss** [comment]
I don't think it is safe to close this issue. I checked the master branch, and there are lots of errors, files with no tests, and low test coverage, see attached file:
[test.txt](https://github.com/pkt-cash/pktd/files/8549178/test.txt)
I started it with `./do --test`. In my opinion, it shouldn't be allowed to commit something if a test is broken. Or at least there should be a stable branch where all tests are running. I didn't tried one of the other branches, but I guess it doesn't look much better there. Is the master branch supposed to be the stable branch used for releases? Then all tests should work there.

Also the test output is very noisy with the syncer and gossiper programs etc. The info and debug messages of these programs should be suppressed in test mode, to see the test results more clearly.

**Comment by Frank-Buss** [comment]
Instead of fixing all broken tests, they should be at least disabled. Then the remaining tests are useful for regression testing. Because now there are 91 times "FAIL" in the output. How can you find a new FAIL, if you change some code, and it breaks a different test? Of course, all disabled tests should be fixed before the next release, or removed if not relevant anymore.
